### ISSUE TITLE
Ever Oasis - freezes when creating a save file

### ISSUE THREAD

**Issue by Hexagon12** [citra-emu/citra#2802]
Using: Citra Nightly, hash - cac1133 .

The game freezes up when creating a new save file. The screen turns white and throws unmapped Read32 errors. 

How to reproduce: Load up the game, press A and then try to create a save file.

Specs: Intel Core i3-6100, Nvidia GeForce GTX 750 Ti, 6 GB RAM

Log file right here: https://pastebin.com/9snDKWs1

**Comment by yuukiw** [comment]
tried adding a fresh system archive in case and tried several saves from a 3ds.
With a save it will always crash when loading said saves.

**Comment by Marborao** [comment]
Any progress? Can someone give some love to this game? thank you.

**Comment by Jhon591** [comment]
New update from me drwhojan here, interesting..
https://gbatemp.net/threads/citra-unofficial-chinese-builds-discussion.431974/page-310#post-7658005

**Comment by Jhon591** [comment]
Edit: seems to be some sort of Down Count.

**Comment by acnleditor2** [comment]
@Hexagon12 test with #3091

**Comment by FearlessTobi** [comment]
Just tested it. The freeze doesn´t seem to be fixed.

**Comment by acnleditor2** [comment]
@FearlessTobi game region and citra region?

**Comment by FearlessTobi** [comment]
Europe and Europe. But why should that be important in any way?

**Comment by acnleditor2** [comment]
freeze with europe and auto

**Comment by bakugo** [comment]
Using the latest commit https://github.com/citra-emu/citra/pull/3091/commits/df7808c9e134343ada3e31c664759757e563f1fb from https://github.com/citra-emu/citra/pull/3091 fixes this freeze.

![image](https://user-images.githubusercontent.com/6014513/32592419-7d4f907e-c51b-11e7-9b07-f0202c40810d.png)

I tried loading the save from my 3ds to see how it works ingame: the screen is entirely black, but I can hear the game running in the background and my character moving when I press buttons. I'll test it some more later today.

**Comment by PPLToast** [comment]
Can confirm here it works too, though yeah you have to pause and unpause the game to see the screen. Strange bug.

**Comment by PananaPhone** [comment]
It appear the game freeze again with same error after long unskippable intro.

**Comment by bakugo** [comment]
It only freezes if you had an existing save and try to start the game with it. Delete the save before starting and it should work. Protip: disable vsync and framerate limit to get past the cutscene pretty fast.

Though it looks like many things don't trigger properly after the game starts (when you first go ingame a conversation should trigger before you can walk around, but that doesn't happen, and other things needed to progress don't work either) so it's impossible to start.

**Comment by asbel123** [comment]
there new build all the games is fullspeed and fullhd u can play shin men tensi 4 fullspeed and kingdom hearts dd hd

 its called citra sickc build get it from youtube u want to play everygame fullspeed even dragon quest 11 hd even dragon quest 8 is hd now it will go low res on battle and ever osaisa is perefect now there is a hack option on this build

 i wish citra add that compontent

**Comment by B3n30** [comment]
Closed by #3091

**Comment by bakugo** [comment]
This is not fixed in the current nightly, please re-open.
Only the commit with a delay of 54.14 microseconds works.

**Comment by Senjosei** [comment]
Tested on canary 268
If u set the `IPCDelayNanoseconds` to 10000000 it loads correctly with all the triggers but the game runs super slow so i debug outputed the thread names and theres this weird `Path: [Binary: 000000000000000000000000]_Server` . If u only delay the `Path: [Binary: 000000000000000000000000]_Server` it runs smoothly without any errors 
```
if(name == "Path: [Binary: 000000000000000000000000]_Server"){
    static constexpr u64 IPCDelayNanoseconds = 10000000;
    thread->WakeAfterDelay(IPCDelayNanoseconds);
} else {
    static constexpr u64 IPCDelayNanoseconds = 51140;
    thread->WakeAfterDelay(IPCDelayNanoseconds);
}
```

**Comment by valentinvanelslande** [comment]
@Senjosei disable limit framerate?

**Comment by bakugo** [comment]
Before I continue, let me start by saying that I have no idea what I'm doing.

I did some testing, and the only ServerSession that needs to be delayed is one created by `Service::FS::File::Connect` when the game opens its own RomFS (it has the name `Path: [Binary: 000000000000000000000000]_Server` as mentioned by @Senjosei). No others need to be delayed for the game to load properly in all cases I tested. The opened file is 890961920 bytes and the minimum delay I was able to use was 1500000 nanoseconds.

This is probably a stupid question, but isn't this just caused by file read delays being longer on hardware than on Citra?
